shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -1,379 +1,223 @@
- * @param source - The source object which will be assigned / called by setting / calling the targets proxies
- * @param chkSet - An optional callback to determine whether a specific property/function should be proxied
- * @memberof Initialization
+ * Provides a collection of utility functions, included for backward compatibility with previous releases.
+ * @deprecated Marking this interface and instance as deprecated in favor of direct usage of the helper functions
+ * as direct usage provides better tree-shaking and minification by avoiding the inclusion of the unused items 
+ * in your resulting code.
  */
-export function proxyAssign(target: any, source: any, chkSet?: (name: string, isFunc?: boolean, source?: any, target?: any) => boolean) {
-    if (target && source && target !== source && _isObject(target) && _isObject(source)) {
-        // effectively apply/proxy full source to the target instance
-        for (const field in source) {
-            if (CoreUtils.isString(field)) {
-                let value = source[field] as any;
-                if (_isFunction(value)) {
-                    if (!chkSet || chkSet(field, true, source, target)) {
-                        // Create a proxy function rather than just copying the (possible) prototype to the new object as an instance function
-                        target[field as string] = (function(funcName: string) {
-                            return function() {
-                                // Capture the original arguments passed to the method
-                                var originalArguments = arguments;
-                                return source[funcName].apply(source, originalArguments);
-                            }
-                        })(field);
-                    }
-                } else if (!chkSet || chkSet(field, false, source, target)) {
-                    if (_hasOwnProperty(target, field)) {
-                        // Remove any previous instance property
-                        delete target[field];
-                    }
-
-                    if (!objDefineAccessors(target, field, () => {
-                        return source[field];
-                    }, (theValue) => {
-                        source[field] = theValue;
-                    })) {
-                        // Unable to create an accessor, so just assign the values as a fallback
-                        // -- this will (mostly) work for objects
-                        // -- but will fail for accessing primitives (if the source changes it) and all types of "setters" as the source won't be modified
-                        target[field as string] = value;
-                    }
-                }
-            }
-        }
-    }
+export interface ICoreUtils {
 
-    return target;
-}
-
-export class CoreUtils {
-    public static _canUseCookies: boolean;
+    /**
+     * Internal - Do not use directly.
+     * @deprecated Direct usage of this property is not recommend
```

Comment:
More details will be added around this when the CookieManager changes are merged (plan is before next release)